identity/README.md

```diff
@@ -0,0 +1,279 @@
+# Identity Service
+
+This document and the Identity Service is based on the [Container Storage
+Interface specification][csi_spec].
+
+## Terminology
+
+| Term       | Definition                                                                               |
+| ---------- | ---------------------------------------------------------------------------------------- |
+| CSI        | [Container Storage Interface][csi_spec].                                                 |
+| CSI-driver | A component that implements CSI service RPCs, the CSI Controller or NodePlugin.          |
+| CO         | Container Orchestration system that communicates with users and CSI-Addons service RPCs. |
+| SP         | Storage Provider, the vendor of a CSI-driver implementation.                             |
+| RPC        | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).            |
+
+## Objective
+
+Define an API that makes it possible for the CSI-Addons plugin for a CO to
+communicate with CSI-driver components that can execute storage maintenance
+operations.
+
+### Goals in MVP
+
+The API will
+
+* make it possible for worker nodes (CSI NodePligin) to announce themselves
+* have the CSI-Addons plugin for the CO list all announced CSI-Addons capable
+  CSI-drivers
+* provide details about CSI-Addons features the CSI-driver supports
+
+### Non-Goals in MVP
+
+This will be limited to node discovery functionality only. CSI-Addons features
+will be documented in their own specifications. The available capabilities will
+get extended when new CSI-Addons operations are proposed.
+
+## Solution Overview
+
+The CSI-Addons plugin for a CO reacts on events generated by the CO on request
+of user interactions. Events are translated to CSI-Addons gRPC procedures which
+are relayed to the CSI-drivers (CSI Controller and NodePlugin).
+
+## Architecture
+
+A user (someone that creates/consumes CSI volumes) instructs the CSI-Addons
+plugin of the CO to invoke a storage maintenance operation. This interaction is
+done in a CO specific way (e.g. a Custom Resourse will be provided for
+Kubernetes).
+
+The CSI-Addons plugin for the CO inspects the instruction from the user, and
+formats it into an operation that is sent to the CSI-driver Controller and/or
+NodePlugin.
+
+```text
+.------.       .------------.
+| User |-------| CSI-Addons |
+'------'       | CO-plugin  |
+               '------------'
+                      |
+                      | gRPC
+                      |
+               .------------.
+               | CSI-driver |
+               '------------'
+```
+
+## RPC Interface
+
+**Identity Service**: The CSI-driver **MUST** implement this, the CO=plugin
```

Comment:
CO=plugin? Do you mean CO here is a plugin?

Reply:
No, a _CO-plugin_. For Kubernetes this will be implemented as a CSI-driver sidecar (and a CSI-Addons Controller that talks to all the CSI-Addons sidecars).